visualization/app/codeCharta/state/store/metricData/nodeMetricData/nodeMetricData.actions.ts

```diff
@@ -0,0 +1,36 @@
+import { Action } from "redux"
+import { BlacklistItem, NodeMetricData } from "../../../../codeCharta.model"
+import { FileState } from "../../../../model/files/files"
+
+export enum NodeMetricDataActions {
+	SET_NODE_METRIC_DATA = "SET_NODE_METRIC_DATA",
+	CALCULATE_NEW_NODE_METRIC_DATA = "CALCULATE_NEW_NODE_METRIC_DATA"
+}
+
+export interface SetNodeMetricDataAction extends Action {
+	type: NodeMetricDataActions.SET_NODE_METRIC_DATA
+	payload: NodeMetricData[]
+}
+
+export interface CalculateNewNodeMetricDataAction extends Action {
+	type: NodeMetricDataActions.CALCULATE_NEW_NODE_METRIC_DATA
+	payload: { fileStates: FileState[]; blacklist: BlacklistItem[] }
+}
+
+export type NodeMetricDataAction = SetNodeMetricDataAction | CalculateNewNodeMetricDataAction
+
+export function setNodeMetricData(nodeMetricData: NodeMetricData[] = defaultNodeMetricData): SetNodeMetricDataAction {
+	return {
+		type: NodeMetricDataActions.SET_NODE_METRIC_DATA,
+		payload: nodeMetricData
+	}
+}
+
+export function calculateNewNodeMetricData(fileStates: FileState[], blacklist: BlacklistItem[]): CalculateNewNodeMetricDataAction {
```

Comment:
This function does not calculate anything. Thus, the name seems rather confusing.

Reply:
action creator, explained in another comment

Reply:
This is really confusing for someone like me who is not that involved with the codebase :)

Reply:
That is true. New developers are aware of the framework redux and are [recommended to read](https://maibornwolff.github.io/codecharta/docs/new-to-code/#state-management) the core concepts of it. [Action creators](https://redux.js.org/basics/actions) are explained in there.